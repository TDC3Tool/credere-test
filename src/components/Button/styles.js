import styled from 'styled-components'

export const ButtonElement = styled.button`
  cursor: pointer;
  background-color: #f60d02;
  box-shadow: inset 0 -1px 0 rgba(0,0,0,0.1);
  border: 0;
  padding: 12px 22px;
  color: #ffffff;
  border-radius: 4px;
  transition: All 0.3s ease-in;

  &:hover {
    box-shadow: inset 0 -3px 0 rgba(0,0,0,0.1);
  }

  &:active {
    box-shadow: inset 0 0 0 transparent;
    transform: translateY(1px);
  }
`
