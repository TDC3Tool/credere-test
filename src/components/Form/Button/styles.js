import styled from 'styled-components'
import { successColor, defaultColor, defaultRadius } from '../../../globalStyles'

export const ButtonElement = styled.button`
  cursor: pointer;
  background-color: transparent;
  box-shadow: inset 0 -1px 0 rgba(0,0,0,0.1);
  border: 1px solid transparent;
  border-radius: ${defaultRadius};
  transition: All 0.3s ease-in;

  &:hover {
    box-shadow: inset 0 -3px 0 rgba(0,0,0,0.1);
  }

  &:active {
    box-shadow: inset 0 0 0 transparent;
    transform: translateY(1px);
  }

  &.button-normal-default {
    background-color: ${defaultColor};
    color: #ffffff;
  }

  &.button-normal-success {
    background-color: ${successColor};
    color: #ffffff;
  }

  &.button-outline-default {
    color: ${defaultColor};
    border-color: ${defaultColor};
  }

  &.button-outline-success {
    color: ${successColor};
    border-color: ${successColor};
  }

  &.button-lg {
    padding: 12px 22px;
  }

  &.button-md {
    padding: 8px 18px;
  }

  &.button-md {
    padding: 6px 16px;
  }
`
