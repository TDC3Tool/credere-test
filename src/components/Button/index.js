import React from 'react'
import PropTypes from 'prop-types'
import { ButtonElement } from './styles'

const Button = ({ children, onClick }) => (
  <ButtonElement
    className="transition-all"
    type="button"
    onClick={onClick}
  >
    {children}
  </ButtonElement>
)

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
}

Button.defaultProps = {
  onClick: () => {},
}

export default Button
