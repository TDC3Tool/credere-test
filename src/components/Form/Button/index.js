import React from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'
import { ButtonElement } from './styles'

const Button = ({
  children, onClick,
  size, modifier, theme,
  ...props
}) => (
  <ButtonElement
    className={classnames(
      `button-${size}`,
      `button-${modifier}-${theme}`,
      'transition-all',
    )}
    onClick={onClick}
    {...props}
  >
    {children}
  </ButtonElement>
)

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  type: PropTypes.string,
  size: PropTypes.oneOf(['lg', 'md', 'sm']),
  modifier: PropTypes.oneOf(['outline', 'normal']),
  theme: PropTypes.oneOf(['default', 'success']),
}

Button.defaultProps = {
  size: 'lg',
  modifier: 'normal',
  type: 'button',
  theme: 'default',
  onClick: () => {},
}

export default Button
