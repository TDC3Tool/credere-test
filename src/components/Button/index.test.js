import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Button from '.'

describe('Button Component', () => {
  it('should render button component', () => {
    const element = renderer.create(<Button>Default Button</Button>).toJSON()
    expect(element).toMatchSnapshot()
  })

  it('should match component type', () => {
    const element = renderer.create(<Button>Default Button</Button>).toJSON()
    expect(element).toMatchObject({
      type: 'button',
    })
  })

  it('should do button action', () => {
    const mockCallBack = jest.fn()
    const element = mount(<Button onClick={mockCallBack}>Do!</Button>).find('button')
    element.simulate('click')
    expect(mockCallBack.mock.calls.length).toEqual(1)
  })
})
