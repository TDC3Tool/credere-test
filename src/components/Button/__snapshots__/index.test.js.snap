// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render button component 1`] = `
<button
  className="transition-all sc-bdVaJa eMbgWz"
  onClick={[Function]}
  type="button"
>
  <span
    className="sc-ifAKCX cFlEyZ"
    style={
      Object {
        "color": "#FFFFFF",
      }
    }
  >
    Default Button
  </span>
</button>
`;
